### PR TITLE
Migrate e2e sythetics 2 GH Actions Workflow

### PR DESCRIPTION
**Pull request by reakaleek**
## Details

Migrate e2e synthetics mbp and schedule to GH Actions workflow

## Motivation

Jenkins to GH Actions Migration

### CODE REVIEW

**Review by amannocci — APPROVED** [review]
LGTM
We can wait for the following [PR](https://github.com/elastic/synthetics/pull/728) to be merged and create a common `tests-reporter` workflow.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  workflow_dispatch: ~
+  schedule:
+    - cron: "0 4,16 * * 1-5"
+  push:
+    branches:
+      - main
+
+jobs:
+  integration-test:
```
Comment (v1v) [inline .github/workflows/e2e.yml:12]: NIT

```suggestion
  e2e-test:
```

File: .github/workflows/e2e-test-reporter.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+## Workflow to process the JUnit test results and add a report to the checks.
+name: e2e-test-reporter
```
Comment (v1v) [inline .github/workflows/e2e-test-reporter.yml:3]: What's the plan for using one test-reporter workflow instead?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/e2e.yml:12]: The thing is.. in the other workflow in #728 there is a job called e2e-test aswell and I wanted to make it distinct.

In this case the full status check will ne named `e2e / Integration-test`

File: .github/workflows/e2e-test-reporter.yml (hunk 2 above)
Comment (reakaleek) [reply .github/workflows/e2e-test-reporter.yml:3]: When #728 is merged. I will modify it.

I was thinking about using the same `test-reporter` workflow and based on the workflow_run name change the test report name or something. But let's see when I'm at that point.

**Review by v1v — APPROVED** [review]
LGTM, just a tiny minor comment regarding the job name

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/e2e.yml]: https://github.com/elastic/synthetics/pull/730/commits/3d17d9665e656983059bdaaa8f3f9793e337bdf2

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

File: .github/workflows/test-reporter.yml
Comment (reakaleek) [inline .github/workflows/test-reporter.yml:1]: @v1v FYI, my attempt to have multiple workflows in test-reporter

Comment (reakaleek) [reply .github/workflows/test-reporter.yml:1]: just to make sure there is no unintentional side effect when the event is triggered twice by different workflows

**Pull request merged**